### DISCUSSION TITLE
issue: `id` is missing on `form.register` return type

### DISCUSSION THREAD

**Discussion by dawsonc623** [react-hook-form/react-hook-form#10930] [Q&A]
### Version Number

7.45.4

### Codesandbox/Expo snack

N/A

### Steps to reproduce

This is an issue with the types, not a bug during execution. The type `UseFormRegister` does not include an `id` property, but function returns one as part of the object it returns.

### Expected behaviour

The return type for `form.register` accurately represents what is returned.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
`register`'s not supposed to return `id`; it returns `name`. Can you reproduce the behavior you're seeing in a codesandbox?

**Comment by dawsonc623** [comment]
For some reason my company has blocked codesandbox on the network. I will loop back around and create a minimal repro when I get home later.

**Comment by dawsonc623** [comment]
I have not been successful in reproducing what I am seeing (an `id` appears on form elements despite not being set directly) in CodeSandbox and also noticed the behavior is inconsistent in the offending codebase. I was pointed to `react-hook-form` as the "ground zero", but now I am thinking it is coming from somewhere else.

**Comment by dawsonc623** [comment]
Sorry to leave this in the dark for so long - I got pulled away for a bit and just recently was able to get back to this. I discovered the underlying source for `id` was not directly from this library as I was pointed toward, but rather there was some mysticism in the project that it seemed no one on the team was aware of. I apologize for the false alarm.
